typescript/cli/tests/warp.zs-test.ts

```diff
@@ -0,0 +1,138 @@
+import { JsonRpcProvider } from '@ethersproject/providers';
+import { Wallet } from 'ethers';
+import { $ } from 'zx';
+
+import { WarpCoreConfig, WarpRouteDeployConfig } from '@hyperlane-xyz/sdk';
+import { Address, assert, eqAddress } from '@hyperlane-xyz/utils';
+
+import { HypNative__factory } from '../../../solidity/dist/factories/contracts/token/HypNative__factory.js';
+import { readYamlOrJson, writeYamlOrJson } from '../src/utils/files.js';
+
+/// To run: 1) start an anvil, 2) yarn run tsx tests/warp.zs-test.ts inside of cli/
+
+const BURN_ADDRESS = '0x0000000000000000000000000000000000000001';
+const LOCAL_ANVIL_HOST = 'http://localhost:8545';
+const CHAIN_NAME = 'anvil1';
+const EXAMPLES_PATH = './examples';
+const TEST_CONFIGS_PATH = './test-configs';
+const REGISTRY_PATH = `${TEST_CONFIGS_PATH}/anvil`;
+const CORE_CONFIG_PATH = `${EXAMPLES_PATH}/core-config.yaml`;
+const WARP_CONFIG_PATH = `${EXAMPLES_PATH}/warp-route-deployment.yaml`;
+const WARP_CORE_CONFIG_PATH = `${REGISTRY_PATH}/deployments/warp_routes/ETH/anvil1-config.yaml`;
+
+const ANVIL_KEY =
+  '0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80';
+
+/**
+ * Deploys the Hyperlane core contracts to the specified chain using the provided config.
+ */
+async function hyperlaneCoreDeploy(coreInputPath: string) {
+  await $`yarn workspace @hyperlane-xyz/cli run hyperlane core deploy \
```

Comment:
In cases like this, the fn need not be async. Just return the promise that $ gives you. The only reason to explicitly unwrap the promise like this is if you need it this way for error handling